### PR TITLE
[Server] Refactoring + Handling OPTIONS + Part 1 of email verification

### PR DESCRIPTION
**Pull request by lancelafontaine**

### CODE REVIEW

File: server/src/controller.py [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
 import rest_service
-
+from helpers import ok, error
 
 def get_heartbeat(request=None):
     return json(ok('hello world'))
 
-
-def get_slack_users(request=None):
+def get_slack_users(request=None, db_session=None):
```
Comment (tamarindd) [inline server/src/controller.py]: why pass db_session to this?

File: server/src/controller.py [hunk 2]
```diff
@@ -23,8 +22,42 @@ def get_slack_users(request=None):
...
+    # - registered: FALSE
+    # - last_attempted_registration: CURRENT DATE
+    # - confirmation_code: VARCHAR (random string)
+
+    # TODO: Make sure that the password is long enough
+    # If so, return here, front-end should refuse registration
+
+    return json(ok(request.json))
+
+def authenticate_user(request=None, db_session=None):
```
Comment (tamarindd) [inline server/src/controller.py]: This method (and all the following ones) won't work if request and db_session are None; they're not really optional and shouldn't have defaults imo

**Comment by jacobrs** [comment]
I'll check this PR after work.

File: README.md [hunk 3]
```diff
@@ -12,4 +12,4 @@
 ### Server
 
 - `python3 -m pip install -r server/requirements.txt`
-- `SLACK_TOKEN=%enter_token_here% python3 server/src/run.py`
+- `SLACK_TOKEN=%enter_token_here% SPARKPOST_TOKEN=%enter_token_here% python3 server/src/run.py`
```
Comment (jacobrs) [inline README.md:15]: Will you send us the sparkpost token? Is it a personal thing or SCS thing?

File: client/src/components/Login.vue [hunk 4]
```diff
@@ -102,6 +102,15 @@ export default {
             formValues[inputs[i].id] = inputs[i].value
           }
         }
+        restFactory.postVerifySlackEmail(formValues).then((res) => {
+          if (res.status === 200) {
+            console.log(res.data.data)
```
Comment (jacobrs) [inline client/src/components/Login.vue]: Console log? Are you planning on changing this in part 2?

File: server/src/controller.py (hunk 1 above)
Comment (jacobrs) [reply server/src/controller.py]: +1

File: server/src/controller.py (hunk 2 above)
Comment (jacobrs) [reply server/src/controller.py]: I agree with Arielle

File: server/src/decorators.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+from functools import wraps
+from sanic.response import json
+from helpers import ok
+
+def responds_to_options():
```
Comment (jacobrs) [inline server/src/decorators.py:5]: This is cool!

**Review by jacobrs — COMMENTED** [review]
Coverage has gone down, we should try and avoid that, hoping to see more tests in Part 2

File: README.md (hunk 3 above)
Comment (lancelafontaine) [reply README.md:15]: It's an SCS thing, and it's pinned in our Slack channel.

File: client/src/components/Login.vue (hunk 4 above)
Comment (lancelafontaine) [reply client/src/components/Login.vue]: Yes, this will be changed. For now it's just there to validate that we are hitting the endpoint.

**Review by tamarindd — APPROVED** [review]

**Pull request merged**